src/Model/ModelTrait.php

```diff
@@ -60,19 +63,19 @@ protected function loadData(array $data, $update = false, $replaceCollections =
                     // if the setter's first parameter is an array, we need to check if this is a collection of objects
                     // in this case there should be an "addProperty" method on the class
                     $adder = "add$prop";
-                    if ($param->isArray() && method_exists($this, $adder)) {
+                    if ($param->getType() && $param->getType()->getName() === 'array' && method_exists($this, $adder)) {
                         $param = $this->getFirstParameter($adder);
                         $isCollection = true;
                     }
                     $value = $this->constructClasses($param, $value, $isCollection);
 
-                    // if we have a collection and we're updating it ...
+                    // if we have a collection, and we're updating it ...
                     if (
                         $isCollection &&
                         $update &&
                         (
                             empty($replaceCollections) ||               // don't add if we're replacing all collections
-                            empty($replaceCollections[$this->mbLcfirst($prop)])  // don't add if this collection has been marked for replacing
+                            empty($replaceCollections[$this->mbLcfirst($prop)])  // don't add if this collection has been)
```

Comment:
Think you might have accidentally removed half this comment :)

Reply:
Restored